src/02-common-principles.md

```diff
@@ -125,6 +125,25 @@ The solutions will change from case to case and publicly available datasets will
 be reviewed to include common data types in the future releases of the BIDS
 specification.
 
+### Metadata conflicts
+
+BIDS specifies that metadata be stored in human-readable formats such as
+TSV (see [Tabular files](#tabular-files)) or
+JSON (see [Key/value files](#keyvalue-files-dictionaries)).
+In some cases, this duplicates metadata contained in a data file.
+
+In cases of conflict, the BIDS metadata is considered authoritative.
```

Comment:
could we clarify what a conflict is? for example, is a change of floating point precision a conflict? one reason we store things in binary is because floating point data maintains some veracity. if we see this outside the nifti lens for other modalities allowed by bids (the eeg/meg or upcoming ephys/microscopy formats, where precision starts mattering).

Reply:
> could we clarify what a conflict is?

I can think of a few:

1) Equivalent fields, different values. `RepetitionTime` vs `pixdim[4]`
2) Disparate representations. `PhaseEncodingDirection` (enum) vs `dim_info` (bitfield)
3) Range of representations. `PhaseEncodingDirection` (axis + direction) vs `dim_info` (axis)

> for example, is a change of floating point precision a conflict?

I'm interpreting this as a question of tolerance? Yes, there should be some tolerance. I imagine the order of tolerance would probably be a case-by-case question.

> one reason we store things in binary is because floating point data maintains some veracity. if we see this outside the nifti lens for other modalities allowed by bids (the eeg/meg or upcoming ephys/microscopy formats, where precision starts mattering).

This seems to be arguing that we should say the highest-precision metadata should be authoritative, whether BIDS or format-internal. Or am I misunderstanding?

Reply:
> This seems to be arguing that we should say the highest-precision metadata should be authoritative, whether BIDS or format-internal.

in general yes, but we may want to clarify that for floating point values, the internal format always wins when close. this would perhaps avoid the situations where some has zeroed out something like TR.

Reply:
This feels like a hard thing to write a rule for. Here's something:

> In general, the more precise metadata representation SHOULD be considered authoritative, with BIDS metadata being presumptively more precise. A notable exception to this presumption is if a value is represented as a floating point value in the internal metadata format and a decimal string in the JSON sidecar; to avoid accumulating error in float-decimal conversions, the floating point value is authoritative.

This actually contravenes existing practice in fMRIPrep and FitLins with regard to `RepetitionTime`/`pixdim[4]`, and may do so for MNE-Python as well, but I guess that's what standardization is for.

Reply:
sounds good to me.

ping @satra

Reply:
i like @effigies paragraph - but i think he still had the question out for the community.

Reply:
@yarikoptic I intended to write it to mean (and still read it as saying) "floating point fields in a binary format take precedence over JSON, which is inherently decimal".

Reply:
to freshen up: re my comment on floating point representation in text. data representation would also be truncated and it would just be a matter of expressing enough precision in json form. 

<details>
<summary>E.g. specifically for my own example messing around in ipython</summary> 

```
In [1]: repr(1/3)
Out[1]: '0.3333333333333333'

In [2]: 0.3333333333333333 * 3
Out[2]: 1.0

In [3]: 3.333333333333333e-1 * 3
Out[3]: 1.0

In [4]: 3.33333333333333e-1 * 3
Out[4]: 0.9999999999999989

In [5]: 1/3 == 3.333333333333333e-1
Out[5]: True

In [6]: 1/3 == 3.33333333333333e-1
Out[6]: False

```
</details>

so may be we should just add somewhere wording that "serialization of floating point numbers in json should be to maximal precision of the underlying data type used in the data file" or something like that?